server/controllers/referral.controllers.js

```diff
@@ -0,0 +1,70 @@
+import {
+  getAllUserReferrals,
+  updateReferralToRewarded,
+  getReferralByEmail,
+  sendReferralInvite,
+} from '../services/referral.service';
+import httpStatus from '../helpers/httpStatus';
+import EMAIL_CONTENT from '../../mailer';
+import { sendMail } from '../services/mailer.service';
+
+const ReferralController = {
+  getAllUserReferrals(req, res, next) {
+    const referrerId = req.params.id;
+    getAllUserReferrals(referrerId)
+      .then((referrals) => {
+        res.status(httpStatus.OK).json({ success: true, referrals });
+      })
+      .catch((error) => next(error));
+  },
+
+  getReferrals(req, res, next) {
+    const { user } = req;
+    getAllUserReferrals(user._id)
+      .then((referrals) => {
+        res.status(httpStatus.OK).json({ success: true, referrals });
+      })
+      .catch((error) => next(error));
+  },
+
+  updateReferralToRewarded(req, res, next) {
+    const { referralId } = req.locals;
+    updateReferralToRewarded(referralId)
+      .then((referral) => {
+        res.status(httpStatus.OK).json({ success: true, message: 'Referral rewarded', referral });
+      })
+      .catch((error) => next(error));
+  },
+
+  getByEmail(req, res, next) {
+    const { email } = req.locals;
+    getReferralByEmail(email)
+      .then((referral) => {
+        res.status(httpStatus.OK).json({ success: true, referral });
+      })
+      .catch((error) => next(error));
+  },
+
+  sendInvite(req, res, next) {
+    const { email } = req.locals;
+    const referrerId = req.user._id;
+    sendReferralInvite({ referrerId, email })
+      .then((inviteInfo) => {
+        const contentTop = `${inviteInfo.referrerName} just invited you to BALLERS.`;
+        const contentBottom = `Use their code ${inviteInfo.referralCode}, or click the link below`;
+        sendMail(
+          EMAIL_CONTENT.REFERRAL_INVITE,
+          { email: inviteInfo.email, firstName: ' ' },
+          {
+            contentTop,
+            contentBottom,
+            link: `http://ballers.ng/referral/invite/${inviteInfo.referralCode}`,
```

Comment:
I believe this should reference the referralId not the referral code to make it easier to load the referral details from the database. 

This means that we will need an endpoint to load a referral via the referralId. 
I believe we can remove referral from the URL to make it shorter.

```suggestion
            link: `http://ballers.ng/invite/${inviteInfo._id}`,
```
See mockup below:

<img width="767" alt="Screenshot 2020-09-20 at 8 12 22 PM" src="https://user-images.githubusercontent.com/26963369/93720345-534a8600-fb80-11ea-9500-693c5b75bf72.png">

The referee name is gotten from the first name and the referrer name is gotten from the user model.

Reply:
For people visiting the site via the referral code that is shared by the user, they will see something like this. We will need another endpoint to cater for them. This endpoint will load the referrer info from the refCode.

<img width="681" alt="Screenshot 2020-09-20 at 8 20 45 PM" src="https://user-images.githubusercontent.com/26963369/93720469-1af77780-fb81-11ea-9a37-a407c048edf3.png">

Reply:
for this, should I create an `invite` route? or it should be appended to the referral route?